src/libraries/System.Private.CoreLib/src/System/Threading/AsyncLocal.cs

```diff
@@ -338,6 +407,12 @@ internal MultiElementAsyncLocalValueMap(int count)
                 _keyValues = new KeyValuePair<IAsyncLocal, object?>[count];
             }
 
+            internal MultiElementAsyncLocalValueMap(KeyValuePair<IAsyncLocal, object?>[] keyValues)
+            {
+                Debug.Assert(keyValues.Length is >= 5 and <= MaxMultiElements);
```

Comment:
Would it be beneficial for `MultiElementAsyncLocalValueMap.Set` to downgrade to a `FourElementAsyncLocalValueMap` when going from 5 to 4 elements?

Reply:
That was the intent, but I neglected to update this check:
https://github.com/dotnet/runtime/pull/68790/files#diff-f7c12a67956e72c9620ce862bff0cfe651e598ec1fa59abea4f7362c54603b1cR439
I'll fix that.